### PR TITLE
Use distinct sign ids when placing col signs

### PR DESCRIPTION
**Pull request by rschifflin**
In Vim, placing or unplacing a sign requires binding an id for that sign. Currently we use the line # of a sign as the id of that sign. This leads us to define the same id multiple times if, for instance, a single line has both a warning and an error. `unplace`ing that sign id 
 (for example, by solving that line's warning but not the error) then incorrectly clears both signs.

This change adds a simple calculation for determining an id for a given Sign. Note that we start numbering at 75000- an [arbitrarily-high number](https://github.com/airblade/vim-gitgutter/blob/master/autoload/gitgutter/sign.vim#L1) to attempt to avoid trampling other plugins' sign ids. We also only `unplace` within the sign's given buffer.

### CODE REVIEW

File: rplugin/python3/LanguageClient/DiagnosticsDisplay.py [hunk 1]
```diff
@@ -1,23 +1,23 @@
 DiagnosticsDisplay = {
-    1: {
+    "1": {
```
Comment (autozimu) [inline rplugin/python3/LanguageClient/DiagnosticsDisplay.py]: Thanks for the work.

Can you explain why this change is necessary?

I am asking because this change would break backward compatibility. User can [customize this display](https://github.com/autozimu/LanguageClient-neovim/blob/master/doc/LanguageClient.txt#L70).

Comment (rschifflin) [reply rplugin/python3/LanguageClient/DiagnosticsDisplay.py]: Ah! I made that change because the integration tests read in the state (including the diagnostics) as a [json payload](https://github.com/autozimu/LanguageClient-neovim/blob/master/rplugin/python3/LanguageClient/LanguageClient_test.py#L47), which only supports string keys, and merges it with the default state containing numbered keys. So the merged end-result DiagnosticsDisplay was 
`{ 1: ..., 2: ..., 3: ..., 4: ..., "1": ..., "2": ..., "3": ..., "4": ... }`
which seemed unintended and threw off the total diagnostic count I was using to calculate sign ids.

I'll simply make a list of the `name`s of a DiagnosticDisplay (excluding duplicates) and use that to calculate an offset instead.

**Comment by autozimu** [comment]
Thanks!
